pkg/iac/terraform/state/backend/http_reader.go

```diff
@@ -27,30 +31,23 @@ func NewHTTPReader(rawURL string, opts *Options) (*HTTPBackend, error) {
 		req.Header.Add(key, value)
 	}
 
-	client := &http.Client{}
-	res, err := client.Do(req)
-	if err != nil {
-		return nil, err
-	}
-
-	buf := new(bytes.Buffer)
-	_, err = buf.ReadFrom(res.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	logrus.WithFields(logrus.Fields{"body": buf.String()}).Trace("HTTP(s) backend response")
-
-	if res.StatusCode < 200 || res.StatusCode >= 400 {
-		return nil, errors.Errorf("error requesting HTTP(s) backend state: status code: %d", res.StatusCode)
-	}
-
-	return &HTTPBackend{rawURL, res.Body}, nil
+	return &HTTPBackend{req, client, nil}, nil
 }
 
 func (h *HTTPBackend) Read(p []byte) (n int, err error) {
 	if h.reader == nil {
-		return 0, errors.New("Reader not initialized")
+		res, err := h.client.Do(h.request)
+		if err != nil {
+			return 0, err
+		}
+		h.reader = res.Body
```

Comment:
I don't think we need the reader to be a struct attribute anymore as we do all the job in the `Read` now

Reply:
What should be the purpose of Close() method then?

Reply:
You are right I forgot the `ReadCloser` interface nevermind